test/helpers/kubectl.go

```diff
@@ -2020,24 +2021,27 @@ func (kub *Kubectl) WaitForKubeDNSEntry(serviceName, serviceNamespace string) er
 		// an IP of the pod itself, not ClusterIPNone, which is what Kubernetes
 		// shows as the IP for the service for headless services.
 		if serviceIP == v1.ClusterIPNone {
-			res := kub.ExecInFirstPod(ctx, LogGathererNamespace, logGathererSelector(false), fmt.Sprintf(digCMD, serviceNameWithNamespace, dnsClusterIP))
+			res := kub.ExecInFirstPod(ctx, LogGathererNamespace, DnsCheckSelector, fmt.Sprintf(digCMD, serviceNameWithNamespace, dnsClusterIP))
 			if res.err != nil {
 				logger.Debugf("failed to run dig in log-gatherer pod")
+				kub.ExecInFirstPod(ctx, LogGathererNamespace, DnsCheckSelector, fmt.Sprintf(digCMDFallback, serviceNameWithNamespace, dnsClusterIP))
 				return false
 			}
-			kub.ExecInFirstPod(ctx, LogGathererNamespace, logGathererSelector(false), fmt.Sprintf(digCMDFallback, serviceNameWithNamespace, dnsClusterIP))
 
-			return res.WasSuccessful()
+			// check whether there is a IP line in dig output
+			ipPresent, _ := hasIPAddress(res.Stdout())
+			return ipPresent
 		}
 		log.Debugf("service is not headless; checking whether IP retrieved from DNS matches the IP for the service stored in Kubernetes")
 
-		res := kub.ExecInFirstPod(ctx, LogGathererNamespace, logGathererSelector(false), fmt.Sprintf(digCMD, serviceNameWithNamespace, dnsClusterIP))
+		res := kub.ExecInFirstPod(ctx, LogGathererNamespace, DnsCheckSelector, fmt.Sprintf(digCMD, serviceNameWithNamespace, dnsClusterIP))
 		if res.err != nil {
 			logger.Debugf("failed to run dig in log-gatherer pod")
 			return false
 		}
-		serviceIPFromDNS := res.SingleOut()
-		if !govalidator.IsIP(serviceIPFromDNS) {
+		hasIPAddress, serviceIPFromDNS := hasIPAddress(res.Stdout())
```

Comment:
I think we have a helper to split by lines, so I don't think you need to split the output in `hasIPAddress`/